src/OrchardCore/OrchardCore.Logging.NLog/WebHostBuilderExtensions.cs

```diff
@@ -18,14 +18,19 @@ public static IWebHostBuilder UseNLogWeb(this IWebHostBuilder builder)
             .ConfigureAppConfiguration((context, _) =>
             {
                 var environment = context.HostingEnvironment;
-                var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);
-                var configDir = string.IsNullOrWhiteSpace(appData) ? $"{environment.ContentRootPath}/{ShellOptionConstants.DefaultAppDataPath}" : appData;
-                LogManager.Configuration.Variables["configDir"] = environment.ContentRootPath;
+                var appData = System.Environment.GetEnvironmentVariable(ShellOptionConstants.OrchardAppData);                
+                var configDir = string.IsNullOrWhiteSpace(appData) ? Path.Combine(environment.ContentRootPath, ShellOptionConstants.DefaultAppDataPath) : appData;
+                if (LogManager.Configuration == null)
+                {
+                    LogManager.Configuration = new LoggingConfiguration();
+                }               
+
+                LogManager.Configuration.Variables["configDir"] = configDir;
             });
     }
 }
 
-internal static class AspNetExtensions
+public static class AspNetExtensions
```

Comment:
This may fail Code Analysis

Reply:
I don't fully understand why this is even made public (or whether it could break code analysis), the linked issue has some hieroglyphs but I don't see relation to this getting public 🤷🏻‍♂️

Reply:
I saw that you originally made the refactor to internal classes. 
I'm asking because I thought it was because of this here:

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1515

It is enabled by default with net9.0. which mean that Code Analysis may fail CI building. At least, I tried with OCC last night and I needed to refactor some classes because Code Analysis is enforced on that repository. It is probably not here, can't remember.

Reply:
I'm refering to this PR here: https://github.com/OrchardCMS/OrchardCore/pull/15668

Reply:
Well this is a static class so the analysis rule probably won't catch this.  Ideally everything internal unless needed to be public in order to avoid breaking change (API that people cannot teference can be changed).

Reply:
If it is internal, where does it be used.

Reply:
> I'm refering to this PR here: #15668
IMHO, if the analysis catch this. i think we should change that analysis. In my memory , at least one guy complain that the OC method change from public to others and his project (or Module, or modified extension) could not continue to work.

Reply:
hmm, as in : should we allow every extension modules to alter the Log configurations. I'm not sure to be honest, we want this to be predictable with every OC website out there. We provide configurations themselves to alter the path of the log files. What would you need else to change? Everything is configurable from NLog.config file...

Let's fix the NRE in this PR and nothing else for now.